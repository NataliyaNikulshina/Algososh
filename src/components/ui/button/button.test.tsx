import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Button", () => {
    it("Корректная отрисовка кнопки с текстом", () => {
      const button = renderer.create(<Button text="Развернуть" />).toJSON();
      expect(button).toMatchSnapshot();
    });

    it("Корректная отрисовка кнопки без текста", () => {
        const button = renderer.create(<Button text="" />).toJSON();
        expect(button).toMatchSnapshot();
      });

    it("Корректная отрисовка задизейбленной кнопки", () => {
        const button = renderer.create(<Button disabled />).toJSON();
        expect(button).toMatchSnapshot();
      });

    it("Корректная отрисовка кнопки с лоудером", () => {
        const button = renderer.create(<Button isLoader />).toJSON();
        expect(button).toMatchSnapshot();
      });

    it("Корректная отрисовка кнопки с типом 'submit' ", () => {
        const button = renderer.create(<Button type="submit" />).toJSON();
        expect(button).toMatchSnapshot();
      });

    it("Корректная отрисовка кнопки с типом 'button' ", () => {
        const button = renderer.create(<Button type="button" />).toJSON();
        expect(button).toMatchSnapshot();
      });

    it("Корректность вызова колбека при клике на кнопку", () => {
        window.alert = jest.fn();
        render(<Button text="Добавить" onClick={() => alert("Тест!")} />);
        const button = screen.getByText("Добавить");
        fireEvent.click(button);
        expect(window.alert).toHaveBeenCalledWith("Тест!");
      });
});