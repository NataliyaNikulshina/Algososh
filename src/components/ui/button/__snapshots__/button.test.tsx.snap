// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Корректная отрисовка задизейбленной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Корректная отрисовка кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Button Корректная отрисовка кнопки с лоудером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Корректная отрисовка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`Button Корректная отрисовка кнопки с типом 'button'  1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Корректная отрисовка кнопки с типом 'submit'  1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="submit"
>
  <p
    className="text undefined"
  />
</button>
`;
